src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -1,19 +1,55 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size = 0;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = indexOf(key);
+        if (index != -1) {
+            values[index] = value;
+        } else {
+            if (size < MAX_SIZE) {
+                keys[size] = key;
+                values[size] = value;
+                size++;
+            }
```

The `put` method does not handle the case where the storage is full. If `size` reaches `MAX_SIZE`, any further attempts to add new elements will silently fail. Consider throwing an exception or resizing the storage when it is full.